File: CashLogApi/CashLogApi/Models/ExpenseCategory.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace CashLogApi.Models
{
    class ExpenseCategory : IEquatable<ExpenseCategory>
    {

        private int _companyId, _categoryId;
        private string _description;

        public int CompanyId { get => _companyId; set => _companyId = value; }
        public int CategoryId { get => _categoryId; set => _categoryId = value; }
        public string Description { get => _description; set => _description = value; }

        public bool Equals(ExpenseCategory other)
        {
            throw new NotImplementedException();
        }
    }
}

File: CashLogApi_VBnet/CashLogLib/Models/ExpenseCategory.vb
Public Class ExpenseCategory
    Implements IEquatable(Of ExpenseCategory)

    Public Property CompanyId As Integer

    Public Property CategoryId As Integer

    Public Property Description As String

    Public Overrides Function Equals(obj As Object) As Boolean
        Return Equals(TryCast(obj, ExpenseCategory))
    End Function

    Public Overloads Function Equals(other As ExpenseCategory) As Boolean Implements IEquatable(Of ExpenseCategory).Equals
        Return other IsNot Nothing AndAlso
               CompanyId = other.CompanyId AndAlso
               CategoryId = other.CategoryId
    End Function

End Class